From eb1542c8e4b12cccc6eaacda4c6d5a8d1b582f7c Mon Sep 17 00:00:00 2001
From: Jouni Malinen <quic_jouni@quicinc.com>
Date: Wed, 31 Jan 2024 12:16:36 +0200
Subject: [PATCH] tests: Close wpa_supplicant control interface sockets at the
 end

Close all the control interface sockets and delete the client socket
files explicitly at the end of the test loop. This removes needs for
various workarounds that tried to force WpaSupplicant and Ctrl class
__del__() to remove the sockets.

Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 tests/hwsim/run-tests.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/hwsim/run-tests.py b/tests/hwsim/run-tests.py
index 1a23e0add..1fdd923c4 100755
--- a/tests/hwsim/run-tests.py
+++ b/tests/hwsim/run-tests.py
@@ -721,6 +721,10 @@ def main():
         if not reset_ok:
             print("Terminating early due to device reset failure")
             break
+
+    for d in dev:
+        d.close_ctrl()
+
     if args.stdin_ctrl:
         set_term_echo(sys.stdin.fileno(), True)
 
-- 
2.43.0

